src/client/components/Maps/MapObjectInfo.js

```diff
@@ -70,11 +81,16 @@ class MapObjectInfo extends React.Component {
 
   getOverlayLayout = () => {
     const { infoboxData } = this.state;
-    const { usedLocale } = this.props;
+    const { usedLocale, isFullscreenMode } = this.props;
     const wobj = getClientWObj(infoboxData.wobject, usedLocale);
     return (
       <Overlay anchor={this.state.infoboxData.coordinates} offset={[-12, 35]}>
-        <div role="presentation" className="MapOS__overlay-wrap" onMouseLeave={this.closeInfobox}>
+        <div
+          role="presentation"
+          className="MapOS__overlay-wrap"
+          onMouseLeave={this.closeInfobox}
+          onClick={isFullscreenMode ? this.toggleModal : null}
```

Comment:
rename 'toggleModal' -> 'closeModal' and check `isFullscreenMode` in this function